bindings/go/constructor/construct.go

```diff
@@ -29,20 +29,25 @@ import (
 // e.g. because the component version already exists in the target repository.
 var ErrShouldSkipConstruction = errors.New("should skip construction")
 
+const AttributeDescriptor string = "constructor/descriptor"
+
 type Constructor interface {
 	// Construct processes a component constructor specification and creates the corresponding component descriptors.
 	// It validates the constructor specification and processes each component in topological order.
-	Construct(ctx context.Context, constructor *constructor.ComponentConstructor) ([]*descriptor.Descriptor, error)
-}
+	Construct(ctx context.Context) error
```

Comment:
nit: is this protected against a double call? (i.e. if discover graph is empty?)